src/gui/modals/ExportProjectDialog.cpp

```diff
@@ -247,10 +270,31 @@ void ExportProjectDialog::onFileFormatChanged(int index)
 	checkBoxVariableBitRate->setVisible(variableBitrateVisible);
 
 	depthWidget->setVisible(bitDepthControlEnabled);
+		
+	// Updating target file extention and checking status
+
+	m_targetFileInfo.setFile(removeKnownExtension(m_targetFileInfo.absoluteFilePath()) 
+							+ ProjectRenderer::fileEncodeDevices[index].m_extension);
+	
+	m_needCheckIfTargetFileExists = m_originalTargetFileExt.compare(ProjectRenderer::fileEncodeDevices[index].m_extension, 
+																	Qt::CaseInsensitive) != 0;
+
+	updateWindowTitle();
 }
 
 void ExportProjectDialog::startBtnClicked()
 {
+	if (m_needCheckIfTargetFileExists && m_targetFileInfo.exists())
+	{
+		QMessageBox mb;
+		mb.setWindowTitle(windowTitle());
+		mb.setText(m_targetFileInfo.fileName() + tr(" already exists. Do you want to replace it?"));
```

Comment:
I realise this is copied from existing code in the project, but gluing strings together like this is bad for translatability. Different languages have considerably different word orders, and the file name won't always be first in the sentence. It would be better to have a placeholder in the string and substitute the file name after translation.

Reply:
I need to correct the translation with addition of a %1 for an argument (file name). I coded it but should I modify all ts files with this new translation key ?

Reply:
The TS file entries are specific to the classes the strings come from. This string is new to this class, so doesn't currently exist in the TS files - the string already there is for `VersionedSaveDialog` instead. Unless you touch that class too (which is out of scope here), there's nothing to change. Even if there were, we don't typically modify TS files when strings change - those are handled separately. (The exception to this is when class names are changed.)